qiskit/visualization/pass_manager_visualization.py

```diff
@@ -0,0 +1,147 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Visualization function for a pass manager. Passes are grouped based on their
+flow controller, and coloured based on the type of pass.
+"""
+import inspect
+from qiskit.transpiler.basepasses import AnalysisPass, TransformationPass
+DEFAULT_STYLE = {AnalysisPass: 'red',
+                 TransformationPass: 'blue'}
+
+
+try:
+    import subprocess
+    print(subprocess.check_output(['dot', '-V']))
+    HAS_GRAPHVIZ = True
+except FileNotFoundError:
+    # this is raised when the dot command cannot be found, which means GraphViz
+    # isn't installed
+    HAS_GRAPHVIZ = False
+
+
+def pass_manager_drawer(pass_manager, filename, style=None):
+    """
+    Draws the pass manager.
+
+    This function needs `pydot <https://github.com/erocarrera/pydot>`, which in turn needs
+    Graphviz <https://www.graphviz.org/>` to be installed.
+
+    Args:
+        pass_manager (PassManager): the pass manager to be drawn
+        filename (str): file path to save image to
+        style (dict or OrderedDict): keys are the pass classes and the values are
+            the colors to make them. An example can be seen in the DEFAULT_STYLE. An ordered
+            dict can be used to ensure a priority coloring when pass falls into multiple
+            categories. Any values not included in the provided dict will be filled in from
+            the default dict
+
+    Raises:
+        ImportError: when nxpd or pydot not installed.
+    """
+
+    try:
+        import pydot
+        if not HAS_GRAPHVIZ:
+            raise ImportError
+    except ImportError:
+        raise ImportError("pass_manager_drawer requires pydot and graphviz. "
+                          "Run 'pip install pydot'. "
+                          "Graphviz can be installed using 'brew install graphviz' on Mac"
+                          " or by downloading it from the website.")
+
+    passes = pass_manager.passes()
+
+    if not style:
+        style = DEFAULT_STYLE
+
+    # create the overall graph
+    graph = pydot.Dot()
+
+    # identifiers for nodes need to be unique, so assign an id
+    # can't just use python's id in case the exact same pass was
+    # appended more than once
+    node_id = 0
+
+    prev_node = None
+
+    for controller_group in passes:
+
+        # label is the name of the flow controller (without the word controller)
+        label = controller_group['type'].__name__.replace('Controller', '')
+
+        # create the subgraph for this controller
+        subgraph = pydot.Cluster(str(id(controller_group)), label=label)
+
+        for pss in controller_group['passes']:
```

Comment:
nitpick: when I came across `pss`, I thought it was an abbreviation, but it's a pass, right? `pass` might be more readable, unless this name is regularly used elsewhere

Reply:
The [preferred way](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) of avoiding collision with Python built-ins is to use a trailing underscore: `pass_`